packages/react-api/src/api/model.js

```diff
@@ -52,13 +54,21 @@ export function executeModel(props) {
     filtersLogicalOperator
   };
 
+  if (spatialFilter) {
+    queryParams.spatialFilter = JSON.stringify(spatialFilter);
+  }
+
   const isGet = url.length + JSON.stringify(queryParams).length <= URL_LENGTH;
   if (isGet) {
     url += '?' + new URLSearchParams(queryParams).toString();
   } else {
+    // undo the JSON.stringify, @todo find a better pattern
```

Comment:
You can't get much better because you have to measure size of `GET` url few lines above and that must have proper url as input.

(btw, it should create proper URL first, measure and test and not rely of more less accourate assumptions that `JSON.stringify(queryParams).` is same length as `new URLSearchParams(queryParams).toString()` but that's other story)

This is done correctly in `getStats` albeit it's not much more readable.

Reply:
I'll try to introduce an utility function for this, in a later PR.